File: app/src/main/java/com/github/dedis/student20_pop/model/network/Unsubscribe.java
package com.github.dedis.student20_pop.model.network;

public final class Unsubscribe extends Request {

    public Unsubscribe(String channel, int id) {
        super(channel, id);
    }

    @Override
    public String getMethod() {
        return Method.UNSUBSCRIBE.getMethod();
    }
}

File: be2-scala/src/main/scala/ch/epfl/pop/model/network/method/Unsubscribe.scala
package ch.epfl.pop.model.network.method

import ch.epfl.pop.json.HighLevelProtocol._
import ch.epfl.pop.model.network.Parsable
import ch.epfl.pop.model.objects.Channel
import spray.json._

final case class Unsubscribe(override val channel: Channel) extends ParamsWithChannel(channel)

object Unsubscribe extends Parsable {
  def apply(channel: Channel): Unsubscribe = {
    new Unsubscribe(channel)
  }

  override def buildFromJson(payload: String): Unsubscribe = payload.parseJson.asJsObject.convertTo[Unsubscribe]
}
